client/windows/GUIClasses.cpp

```diff
@@ -1129,6 +1129,12 @@ CHillFortWindow::CHillFortWindow(const CGHeroInstance * visitor, const CGObjectI
 	statusbar = CGStatusBar::create(std::make_shared<CPicture>(background->getSurface(), Rect(8, pos.h - 26, pos.w - 16, 19), 8, pos.h - 26));
 
 	garr = std::make_shared<CGarrisonInt>(Point(108, 60), 18, Point(), hero, nullptr);
+
+	if(object->typeName == "miniHillFort")
+		statusbar->write(VLC->generaltexth->translate("vcmi.adventureMap.miniHillFort.statusBar.info"));
+	else if(object->typeName == "hillFort")
+		statusbar->write(VLC->generaltexth->translate("vcmi.adventureMap.HillFort.statusBar.info"));
+
```

Comment:
Should be done in config:
- for H3 fort - in `config/objects/generic.json`
- for hota fort - in hota mod.

Game should:
- register this string in HillFortInstanceConstructor::initTypeData
- set this string to HillFort instance HillFortInstanceConstructor::initializeObject

Check CRewardableConstructor for similar code.